go/vt/tabletserver/sqlquery.go

```diff
@@ -175,12 +190,12 @@ func (sq *SqlQuery) disallowQueries() {
 	sq.mu.Unlock()
 	// Terminate all streaming queries
 	sq.qe.streamQList.TerminateAll()
-	// Don't hold lock while waiting.
+	// Wait for outstanding requests to finish.
 	sq.requests.Wait()
 
-	// SHUTTING_QUERIES -> NOT_SERVING
-	sq.mu.Lock()
 	defer func() {
```

Comment:
Well if we panic we have other issues than the mutex, wouldn't this just exit the process?